akka-docs/rst/java/lambda-persistence.rst

```diff
@@ -175,46 +175,29 @@ only be received by a persistent actor after recovery completes.
 Recovery customization
 ^^^^^^^^^^^^^^^^^^^^^^
 
-Automated recovery on start can be disabled by overriding ``preStart`` with an empty implementation.
+Applications may also customise how recovery is performed by returning a customised ``Recovery`` object
+in the ``recovery`` method of a ``PersistentActor``, for example setting an upper bound to the replay,
+which allows the actor to be replayed to a certain point "in the past" instead to its most up to date state:
 
-.. includecode:: ../../../akka-samples/akka-sample-persistence-java-lambda/src/main/java/doc/LambdaPersistenceDocTest.java#recover-on-start-disabled
+.. includecode:: code/docs/persistence/LambdaPersistenceDocTest.java#recovery-custom
 
-In this case, a persistent actor must be recovered explicitly by sending it a ``Recover`` message.
+Recovery can be disabled by returning ``NoRecovery()`` in the ``recovery`` method of a ``PersistentActor``:
```

Comment:
If we go "full in" then yes. Currently it's a factory for `Recovery(0L)`.
If we want to go full in I'll need a trait `Recovery` rename the current case class to something and add that case object.

Current proposal changes name of `case class Recover ` to `Recovery`. It's really misleading when you have `! Recover` and it just doesnt work anymore, so I thought maybe the rename helps with discovering of the change in behaviour as well